python/nano/example/tensorflow/hpo/hpo.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# this file is adapted from the optuna example https://github.com/optuna/optuna-examples/blob/main/keras/keras_simple.py
+
+from tensorflow.keras.datasets import mnist
+from tensorflow.keras.layers import Conv2D, Dense, Flatten
+from tensorflow.keras import Input
+from tensorflow.keras.optimizers import RMSprop
+
+from bigdl.nano.tf.keras import Sequential
+from bigdl.nano.tf.keras import Model
+
+N_TRAIN_EXAMPLES = 3000
+N_VALID_EXAMPLES = 1000
+
+CLASSES = 10
+
+# model_initor takes only a trial argument
+# use lambda to accomodate extra arguments as examples shown in main
+def model_inits(trial, input_shape):
```

Comment:
what is `trial` here?

Reply:
a trial is a tuning instance where user can get suggested value for hyper parameters.

Reply:
i'll try revise the API

Reply:
this example is deprecated